secure-device-storage/sds-android/securedevicestorage/src/main/java/de/adorsys/android/securedevicestorage/KeystoreTool.java

```diff
@@ -0,0 +1,250 @@
+            }
+            return null;
+        }
+    }
+
+    static void deleteKeyPair(@NonNull Context context) throws UnrecoverableKeyException,
+            CertificateException, NoSuchAlgorithmException, KeyStoreException, IOException {
+
+        if (keyPairExists()) {
+            // Delete Key from Keystore
+            getKeyStoreInstance().deleteEntry(KEY_ALIAS);
+        } else {
+            Log.e(KeystoreTool.class.getName(),
+                    context.getString(R.string.message_keypair_does_not_exist));
+        }
+    }
+
+    @RequiresApi(api = Build.VERSION_CODES.ICE_CREAM_SANDWICH)
+    @Nullable
+    static String encryptMessage(@NonNull Context context, @NonNull String plainMessage)
+            throws NoSuchPaddingException, NoSuchAlgorithmException, NoSuchProviderException,
+            IOException, UnrecoverableEntryException, KeyStoreException,
+            CertificateException, InvalidKeyException {
+
+
+        Cipher input;
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.JELLY_BEAN_MR2
+                && Build.VERSION.SDK_INT < M) {
+            input = Cipher.getInstance(KEY_TRANSFORMATION_ALGORITHM, KEY_CIPHER_JELLYBEAN_PROVIDER);
+        } else if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.M) {
+            input = Cipher.getInstance(KEY_TRANSFORMATION_ALGORITHM, KEY_CIPHER_MARSHMALLOW_PROVIDER);
+        } else {
+            return null;
+        }
+        input.init(Cipher.ENCRYPT_MODE, getPublicKey(context));
+
+        ByteArrayOutputStream outputStream = new ByteArrayOutputStream();
+        CipherOutputStream cipherOutputStream = new CipherOutputStream(
+                outputStream, input);
+        cipherOutputStream.write(plainMessage.getBytes("UTF-8"));
+        cipherOutputStream.close();
+
+        byte[] values = outputStream.toByteArray();
+        return Base64.encodeToString(values, Base64.DEFAULT);
+    }
+
+    @Nullable
+    static String decryptMessage(@NonNull Context context, @NonNull String encryptedMessage) throws
+            NoSuchPaddingException, NoSuchAlgorithmException, NoSuchProviderException,
+            IOException, UnrecoverableEntryException, KeyStoreException,
+            CertificateException, InvalidKeyException {
+
+        Cipher output;
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.JELLY_BEAN_MR2
+                && Build.VERSION.SDK_INT < M) {
+            output = Cipher.getInstance(KEY_TRANSFORMATION_ALGORITHM, KEY_CIPHER_JELLYBEAN_PROVIDER);
+        } else if (Build.VERSION.SDK_INT >= M) {
+            output = Cipher.getInstance(KEY_TRANSFORMATION_ALGORITHM, KEY_CIPHER_MARSHMALLOW_PROVIDER);
+        } else {
+            return null;
+        }
+
+        output.init(Cipher.DECRYPT_MODE, getPrivateKey(context));
+
+        CipherInputStream cipherInputStream = new CipherInputStream(
+                new ByteArrayInputStream(Base64.decode(encryptedMessage, Base64.DEFAULT)), output);
+        ArrayList<Byte> values = new ArrayList<>();
```

Comment:
Fixed in commit: https://github.com/adorsys/secure-key-storage/pull/54/commits/6b0ab96771c505e7df50bfa861fdbca6c765a84c